File: backend/src/main/java/com/tfg/backend/model/entity/Movimiento.java
package com.tfg.backend.model.entity;

import com.tfg.backend.auth.models.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "movimientos")
public class Movimiento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "producto_id", nullable = false)
    private Producto producto;

    @ManyToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private User usuario;

    @Enumerated(EnumType.STRING)
    private TipoMovimiento tipo;

    @Column(nullable = false)
    private Integer cantidad;

    @CreationTimestamp
    private Instant createdAt;
    @UpdateTimestamp
    private Instant updatedAt;
}

File: proyecto/frontend/src/app/interfaces/movimiento.ts
import { Producto } from './producto';
import { Usuario } from './usuario';
import { Empresa } from './empresa';

export interface Movimiento {
  id: number;
  tipo: string; // Ejemplo: "ENTRADA" o "SALIDA"
  cantidad: number;
  producto: Producto;
  usuario: Usuario;
  empresa: Empresa;
  createdAt: string;
  updatedAt: string;
}